enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion.tsx

```diff
@@ -0,0 +1,109 @@
+import cx from "classnames";
+import { useEffect } from "react";
+
+import LoadingAndErrorWrapper from "metabase/components/LoadingAndErrorWrapper";
+import CS from "metabase/css/core/index.css";
+import { useDispatch, useSelector } from "metabase/lib/redux";
+import {
+  initializeQB,
+  navigateToNewCardInsideQB,
+  updateQuestion,
+} from "metabase/query_builder/actions";
+import QueryVisualization from "metabase/query_builder/components/QueryVisualization";
+import { FilterHeader } from "metabase/query_builder/components/view/ViewHeader/components";
+import {
+  getCard,
+  getFirstQueryResult,
+  getQuestion,
+  getUiControls,
+} from "metabase/query_builder/selectors";
+import { Group, Stack } from "metabase/ui";
+import { getEmbeddingMode } from "metabase/visualizations/click-actions/lib/modes";
+import type { CardId } from "metabase-types/api";
+
+import { useEmbeddingContext } from "../../context";
+
+interface InteractiveQuestionProps {
+  questionId: CardId;
+}
+
+export const InteractiveQuestion = (
+  props: InteractiveQuestionProps,
+): JSX.Element | null => {
+  const { isInitialized, isLoggedIn } = useEmbeddingContext();
+  const dispatch = useDispatch();
+  const question = useSelector(getQuestion);
+  const mode = question && getEmbeddingMode(question);
+  const card = useSelector(getCard);
+  const result = useSelector(getFirstQueryResult);
+  const uiControls = useSelector(getUiControls);
+
+  const { questionId } = props;
```

Comment:
nit: let's throw the `questionId` in the props like `({questionId}: InteractiveQuestionProps)`. I think that's the direction we're going with most of our TS anyway